Randomly failing test (test_list_worker_slot_warning)

This was seen failing on Travis for Python3.6 when running tests using the django backend.

It was a PR that _only_ touches the documentation (#3335).

Error:

```shell
======================================================================
FAIL: test_list_worker_slot_warning (aiida.backends.tests.cmdline.commands.test_process.TestVerdiProcessListWarning)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/aiidateam/aiida-core/aiida/backends/tests/cmdline/commands/test_process.py", line 414, in test_list_worker_slot_warning
    self.assertClickResultNoException(result)
  File "/home/travis/build/aiidateam/aiida-core/aiida/backends/testbase.py", line 201, in assertClickResultNoException
    self.assertIsNone(cli_result.exception, ''.join(traceback.format_exception(*cli_result.exc_info)))
AssertionError: TypeError('Parser must be a string or character stream, not NoneType',) is not None : Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/testing.py", line 326, in invoke
    cli.main(args=args or (), prog_name=prog_name, **extra)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/travis/build/aiidateam/aiida-core/aiida/cmdline/utils/decorators.py", line 68, in wrapper
    return wrapped(*args, **kwargs)
  File "/home/travis/build/aiidateam/aiida-core/aiida/cmdline/commands/cmd_process.py", line 82, in process_list
    print_last_process_state_change()
  File "/home/travis/build/aiidateam/aiida-core/aiida/cmdline/utils/common.py", line 68, in print_last_process_state_change
    timestamp = get_process_state_change_timestamp(process_type)
  File "/home/travis/build/aiidateam/aiida-core/aiida/engine/utils.py", line 301, in get_process_state_change_timestamp
    timestamps.append(timezone.isoformat_to_datetime(manager.get(key).value))
  File "/home/travis/build/aiidateam/aiida-core/aiida/common/timezone.py", line 129, in isoformat_to_datetime
    return dateutil.parser.parse(value)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/dateutil/parser/_parser.py", line 1358, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/dateutil/parser/_parser.py", line 646, in parse
    res, skipped_tokens = self._parse(timestr, **kwargs)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/dateutil/parser/_parser.py", line 722, in _parse
    l = _timelex.split(timestr)         # Splits the timestr into tokens
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/dateutil/parser/_parser.py", line 207, in split
    return list(cls(s))
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/dateutil/parser/_parser.py", line 76, in __init__
    '{itype}'.format(itype=instream.__class__.__name__))
TypeError: Parser must be a string or character stream, not NoneType
```

--- a/aiida/common/timezone.py
+++ b/aiida/common/timezone.py
@@ -124,6 +124,8 @@ def datetime_to_isoformat(value):
 
     :param value: a datetime object
     """
+    if value is None:
+        return None
     return value.isoformat()
 
 
@@ -132,4 +134,6 @@ def isoformat_to_datetime(value):
 
     :param value: a ISO format string representation of a datetime object
     """
+    if value is None:
+        return None
     return dateutil.parser.parse(value)

